fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,1109 @@
+from concurrent.futures import ThreadPoolExecutor
+import json
+import logging
+import os
+from packaging.requirements import Requirement
+from PIL import Image
+import requests
+import shutil
+
+import huggingface_hub as hfh
```

Comment:
Can you use this import pattern?

```py
hfh = fou.lazy_import(
    "huggingface_hub", callback=lambda: fou.ensure_package("huggingface_hub")
)
```

This will give the user an informative error message if they don't have `huggingface_hub` installed, but also doesn't prevent the module from being imported in case the user is trying to do something that doesn't actually require the `huggingface_hub` package to the installed.